src/lib/rmt/wizard_firewall_page.rb

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'y2firewall/firewalld'
+require 'rmt/utils'
+require 'cwm/dialog'
+require 'cwm/custom_widget'
+
+Yast.import 'CWMFirewallInterfaces'
+
+module RMT; end
+
+class RMT::WizardFirewallPage < CWM::Dialog
+  def initialize
+    textdomain 'rmt'
+  end
+
+  def title
+    _('RMT configuration step 4/5')
+  end
+
+  def abort_button
+    Yast::Label.CancelButton # to be consistent with the other pages
+  end
+
+  def contents
+    if firewalld.installed? && firewalld.enabled?
+      HBox(
+        HStretch(),
+        FirewallWidget.new,
+        HStretch()
+      )
+    else
+      VCenter(
+        Label(_("Firewalld is not enabled.\n\nIf you enable firewalld later,\nremember to open the firewall ports for HTTP and HTTPS."))
+      )
+    end
+  end
+
+  def run
+    read_config
+    result = super
+    write_config if result == :next
+    result
+  end
+
+  # Widget for opening HTTP & HTTPS services in the firewall
+  class FirewallWidget < CWM::CustomWidget
+    attr_accessor :cwm_interfaces
+
+    def initialize
+      textdomain 'rmt'
+
+      @cwm_interfaces = Yast::CWMFirewallInterfaces.CreateOpenFirewallWidget(
+        'services'        => services,
+        'display_details' => true,
+        'open_firewall_checkbox' => _('Open Ports for HTTP and HTTPS in Firewall')
+      )
+    end
+
+    def init
```

Comment:
Why is this separate to initialize?

Reply:
It initializes the widget state:

See https://github.com/yast/yast-yast2/blob/SLE-15-GA/library/cwm/src/lib/cwm/abstract_widget.rb#L46

Reply:
Ask YaST team! ;)
See for example https://github.com/teclator/yast-network/blob/master/src/lib/y2remote/widgets/remote.rb#L174

I don't know the exact life-cycle of their widgets, but this is how they are supposed to be used.

Reply:
@teclator You are too fast, Knut :) Thanks for jumping in!